netplan/cli/sriov.py

```diff
@@ -30,25 +30,36 @@
 def _get_target_interface(interfaces, config_manager, pf_link, pfs):
     if pf_link not in pfs:
         # handle the match: syntax, get the actual device name
-        pf_match = config_manager.ethernets[pf_link].get('match')
+        pf_dev = config_manager.ethernets[pf_link]
+        pf_match = pf_dev.get('match')
         if pf_match:
-            by_name = pf_match.get('name')
-            by_mac = pf_match.get('macaddress')
-            by_driver = pf_match.get('driver')
-
-            for interface in interfaces:
-                if ((by_name and not utils.is_interface_matching_name(interface, by_name)) or
-                        (by_mac and not utils.is_interface_matching_macaddress(interface, by_mac)) or
-                        (by_driver and not utils.is_interface_matching_driver_name(interface, by_driver))):
-                    continue
-                # we have a matching PF
-                # store the matching interface in the dictionary of
-                # active PFs, but error out if we matched more than one
-                if pf_link in pfs:
-                    raise ConfigurationError('matched more than one interface for a PF device: %s' % pf_link)
-                pfs[pf_link] = interface
+            # now here it's a bit tricky
+            set_name = pf_dev.get('set-name')
+            if set_name and set_name in interfaces:
+                # if we had a match: stanza and set-name: this means we should
+                # assume that, if found, the interface has already been
+                # renamed - use the new name
+                pfs[pf_link] = set_name
```

Comment:
I had some problems recently with `set-name:`... it does not seem to always work reliably. The assumption you make here is still valid and needs no changes!

I just want to point out that we should observe the overall `set-name:` functionality in the future and see if there is any debugging or fixes needed.